File: backend/admin-api/adminAPI/Models/UserByType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class UserByType
{
    [Key]
    public int UserID { get; set; }

    public string? UserType { get; set; }

    [Required]
    public string? Username { get; set; }

    public string? DisplayName { get; set; }

    public string? FirstName { get; set; }

    public string? LastName { get; set; }

    public string? UserEmail { get; set; }

    public string? UserContact { get; set; }

   
            
    public string? Images { get; set; }


 

    public string? Description { get; set; }

      public int CreatedBy { get; set; }

}

File: frontend/customer-app/perkup_customer_app/lib/models/user/userbytype.dart
class UserByType {
  final int userID;
  final String? userType;
  final String? username;
  final String? displayName;
  final String? firstName;
  final String? lastName;
  final String? userEmail;
  final String? userContact;
  final String? images;
  final String? description;
  final int createdBy;

  UserByType({
    required this.userID,
    required this.userType,
    required this.username,
    required this.displayName,
    required this.firstName,
    required this.lastName,
    required this.userEmail,
    required this.userContact,
    required this.images,
    required this.description,
    required this.createdBy,
  });

  factory UserByType.fromJson(Map<String, dynamic> json) {
    return UserByType(
      userID: json['userID'],
      userType: json['userType'],
      username: json['username'],
      displayName: json['displayName'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      userEmail: json['userEmail'],
      userContact: json['userContact'],
      images: json['images'],
      description: json['description'],
      createdBy: json['createdBy'],
    );
  }
}
